packages/graphback-runtime/src/utils/fieldTransformHelpers.ts

```diff
@@ -0,0 +1,54 @@
+import { parseMarker } from "graphql-metadata";
+import { GraphQLObjectType } from 'graphql';
+
+export type FieldTransformer = (value?: any) => any;
+
+
+export interface FieldTransform {
+    fieldName: string,
+    transform: FieldTransformer
+}
+
+export enum TransformType {
+    UPDATE = "onUpdateFieldTransform",
+    CREATE = "onCreateFieldTransform"
+}
+
+export type FieldTransformMap = {
+    [TransformType.CREATE]: FieldTransform[],
+    [TransformType.UPDATE]: FieldTransform[]
+};
+
+export function getFieldTransformations(baseType: GraphQLObjectType): FieldTransformMap {
```

Comment:
This function should probably handle `@versioned` marker too.

Reply:
I think this works differently. Versioned is just used at generation phase and it is adding 2 new annotations to target schema. If we were about to add versioned here we could remove other markers like createdAt etc.

Reply:
Would love to help, are you using schema plugin? We are handling `@versioned` in the schema plugin:
https://github.com/aerogear/graphback/blob/06ea54ebd4650586a6d20830b467fdf687ddd676/packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts#L287-L322

Reply:
Yes, OVP is using the schema plugin https://github.com/aerogear/OpenVolunteerPlatform/blob/master/platform/server/.graphqlrc.yml#L17

Reply:
> I think this works differently. Versioned is just used at generation phase and it is adding 2 new annotations to target schema. If we were about to add versioned here we could remove other markers like createdAt etc.

Yes, I expected to have the `createdAt` and `updatedAt` fields to be generated on the fly (and persisted, which is the current bug I am seeing), with the usage of `@versioned`, at least from this is what I see from the code shared in https://github.com/aerogear/graphback/pull/1301#discussion_r434479544

Reply:
I wanted to bring the behaviour tested here https://github.com/aerogear/graphback/pull/1301/files#diff-5e6f4bf330b4e97ce59c1869385ee32bR235